src/recipe/recipe.controller.ts

```diff
@@ -1,8 +1,14 @@
-import { Controller } from '@nestjs/common';
+import { Controller, Get, Param } from '@nestjs/common';
+import { RecipeDto } from './dto/recipe.dto';
 
 import { RecipeService } from './recipe.service';
 
 @Controller('recipe')
 export class RecipeController {
   constructor(private recipeService: RecipeService) {}
+
+  @Get('/search/:query')
```

Comment:
Usually, the search query is passed as a query string, e.q. `?q=asdasd`

There is nothing wrong with your solution here, but you must remember that when using the dynamic path, you must pay close attention to the order of the routes registered in this controller.

Imagine a scenario when, later on, you have to add an endpoint to search by ingredient name, not a title, on a separate endpoint.

If you'd add it after this one with a path like
```ts
@Get('/search/ingredients/:query')
```
a request with this path can possibly be routed to the `/search/:query` path.
If you have situations like this, it's always good to keep the order from the most specific to the most general endpoints.